src/services/codefixes/importFixes.ts

```diff
@@ -501,6 +501,14 @@ namespace ts.codefix {
                             relativeFileName = getRelativePath(moduleFileName, sourceDirectory);
                         }
 
+                        if (startsWith(relativeFileName, "@types/")) {

```

Comment:
It would be nice to have this code next to `mangleScopedPackage` in `moduleNameResolver.ts`. Maybe call it `getPackageNameFromAtTypesDirectory`.